src/Sharpliner/AzureDevOps/Model/Tasks/Marketplace/AzureCliTask.cs

```diff
@@ -0,0 +1,276 @@
+    /// </summary>
+    [YamlIgnore]
+    public string? Arguments
+    {
+        get => GetString("arguments");
+        init => SetProperty("arguments", value);
+    }
+
+    /// <summary>
+    /// Optional. Use when scriptType = ps || scriptType = pscore. Allowed values: stop, continue, silentlyContinue. Default value: stop.
+    /// Prepends the line $ErrorActionPreference = 'VALUE' at the top of your PowerShell/PowerShell Core script.
+    /// </summary>
+    [YamlIgnore]
+    public PowerShellErrorActionPreference PowerShellErrorActionPreference
+    {
+        get => GetEnum("powerShellErrorActionPreference", PowerShellErrorActionPreference.Stop);
+        init => SetProperty("powerShellErrorActionPreference", value);
+    }
+
+    /// <summary>
+    /// Adds the service principal ID, service principal key or workload identity federation token, and tenant ID of the Azure endpoint you chose to the script's execution environment.
+    /// You can use the servicePrincipalId, servicePrincipalKey or idToken, and tenantId variables in your script.
+    /// </summary>
+    [YamlIgnore]
+    public bool AddSpnToEnvironment
+    {
+        get => GetBool("addSpnToEnvironment", false);
+        init => SetProperty("addSpnToEnvironment", value);
+    }
+
+    /// <summary>
+    /// If this input is false, this task will use its own Azure CLI configuration directory.
+    /// Use this task to run Azure CLI tasks in parallel releases.
+    /// </summary>
+    [YamlIgnore]
+    public bool UseGlobalConfig
+    {
+        get => GetBool("useGlobalConfig", false);
+        init => SetProperty("useGlobalConfig", value);
+    }
+
+    /// <summary>
+    /// Current working directory where the script is run.
+    /// If left blank, this input is the root of the repo (build) or artifacts (release), which is $(System.DefaultWorkingDirectory).
+    /// </summary>
+    [YamlIgnore]
+    public string? WorkingDirectory
+    {
+        get => GetString("workingDirectory");
+        init => SetProperty("workingDirectory", value);
+    }
+
+    /// <summary>
+    /// If this input is true, this task will fail when any errors are written to the StandardError stream.
+    /// Clear the checkbox to ignore standard errors and instead rely on exit codes to determine the status.
+    /// </summary>
+    [YamlIgnore]
+    public bool FailOnStandardError
+    {
+        get => GetBool("failOnStandardError", false);
+        init => SetProperty("failOnStandardError", value);
+    }
+
+    /// <summary>
+    /// Use when scriptType = ps || scriptType = pscore.
+    /// If this input is false, the line if ((Test-Path -LiteralPath variable:\LASTEXITCODE)) { exit $LASTEXITCODE } is appended to the end of your script.
+    /// This will propagate the last exit code from an external command as the exit code of PowerShell.
+    /// Otherwise, the line is not appended to the end of your script.
+    /// </summary>
+    [YamlIgnore]
+    public bool PowerShellIgnoreLASTEXITCODE
```

Comment:
this property and `PowerShellErrorActionPreference` are only relevant when "scriptType = ps || scriptType = pscore".

Consider having extended record classes to help with the validations here where these properties will exist on a more specific implementation that uses powershell